source/slang/hlsl.meta.slang

```diff
@@ -4328,7 +4338,7 @@ matrix<T, N, M> exp(matrix<T, N, M> x)
 
 __generic<T : __BuiltinFloatingPointType>
 __target_intrinsic(hlsl)
-__target_intrinsic(glsl)
+__target_intrinsic(glsl, "exp2(float($0))")
```

Comment:
I don't think this is right, especially if T is `float16_t`. Should be kept as `__target_intrinsic(glsl)`.

Reply:
The problem with exp2 on GLSL is that there is no exp2 for double-type.
What I am trying here is to use float version of it when it uses double-type.
But I didn't realized that __BuiltinFloatingPointType includes float16_t.

Hmmm... I will think about how to fix this.

Reply:
You can convert all the __target_intrinsic into a __target_switch inside the body, there you can do:
```
__target_switch
{
   ...
case glsl:
      if (__isHalf<T>() || __isFloat<T>())
            __intrinsic_asm "exp2($0)";
      else
           __intrinsic_asm "exp2(float($0))`;
}
```